elks/arch/i86/kernel/irq.c

```diff
@@ -132,24 +140,23 @@ void INITPROC irq_init(void)
 {
     flag_t flags;
 
-    init_irq();
+    init_irq();  // PIC initialization
+
+	// TODO: no more need to disable the timer
+	// BEFORE the interrupt initialization
+	// as timer vector is kept as is
 
+	// TODO: move that timer stuff in `timer.c`
+	// to clearly separate the PIC and PIT driving
     disable_timer_tick();
```

Comment:
I suggest waiting to next PR on this. Possible no need at all for disable timer function in ELKS, given new way of handling interrupt vectors.

Reply:
Again only an idea, not changing this for now.